### DISCUSSION TITLE
MQTT X on GCP IOT CORE pub/sub

### DISCUSSION THREAD

**Discussion by gylim78** [emqx/MQTTX#930] [Q&A]
Hi everyone,

Just wondering if anyone has managed to connect MQTTX to GCP IOT CORE using the MQTT gateway? 
I tried following the guide on https://cloud.google.com/iot/docs/how-tos/mqtt-bridge#downloading_mqtt_server_certificates

But I am getting "Error: Connection refused: Bad username or password"

I used https://jwt.io/ to create the JWT token as the password by putting in the public and private key.

Has anyone ever got it done? Or is it even possible?

**Comment by ysfscream** [comment]
@gylim78 Sorry for my later reply. I just saw this issue... I'm not sure how the GCP IOT CORE is configured for JWT, for EMQX you can verify the username or password, and once configured, just fill in the JWT token for the username or password and you're done!

**Reply by ysfscream** [reply]
Ref: https://www.emqx.io/docs/en/v4.4/advanced/auth-jwt.html
